python/pyarrow/table.pxi

```diff
@@ -1669,26 +1669,44 @@ def table(data, names=None, schema=None, metadata=None):
             "Expected pandas DataFrame, python dictionary or list of arrays")
 
 
-def concat_tables(tables):
+def concat_tables(tables, c_bool promote=False, MemoryPool memory_pool=None):
     """
-    Perform zero-copy concatenation of pyarrow.Table objects. Raises exception
-    if all of the Table schemas are not the same
+    Perform zero-copy concatenation of pyarrow.Table objects.
```

Comment:
Nit: in the `promote=True` case it isn't quite zero copy case any more. Not sure how to clarify this without suggesting that there is allocation in the `promote=False` case